bridgekeeper/rule_r_tests.py

```diff
@@ -95,6 +96,22 @@ def test_traverse_fk_user_func():
     assert shrubbery_nomatch not in qs
 
 
+def test_traverse_nonexistent_fk():
+    """Comparing with a reverse FK traversal that does not exist for the model."""
+    user = UserFactory(profile=None)
+    profile = ProfileFactory(user=UserFactory(profile=None))
+    user_has_profile = R(profile=profile)
+    user_has_no_profile = R(profile=None)
+
+    # filter() tests
+    queryset = user_has_profile.filter(user, User.objects.all())
+    assert user not in queryset
```

Comment:
I think this is probably missing:

1. A check that `profile.user` is in the queryset. (and just in case this is causing confusion, the `user` being passed to `filter()` here is only used if the `Rule` has a callable on the RHS somewhere. Otherwise I think it's irrelevant and it's the queryset and how it's filtered that matters. Hence the suggestion to check that one is in the filtered queryset and the other is not)
2. Also check `user_has_no_profile.filter(user, User.objects.all())` and assert that `user` is in the queryset and `profile.user` is not (the reverse of the existing checks)

Reply:
Aha! Looks like I got distracted half way through and called it a day!

Reply:
Agreed on `user` being irrelevant. I had to use a pretty contrived rule to fit this within the existing database structure, and I didn't want to change the database structure because that'd necessitate coming up with some more contrived business logic.